ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/StandaloneExecutor.java

```diff
@@ -115,36 +169,86 @@ private void showWelcomeMessage() {
+      final Statement statement,
+      final String statementString,
+      final Map<String, Object> configProperties) {
+    final Query query = getQueryFromStatement(statementString, statement);
+    final QueryMetadata queryMetadata =
+        ksqlEngine.getQueryExecutionPlan(query, ksqlConfig);
+    if (statement instanceof CreateAsSelect) {
+      validateCsasCtas(queryMetadata, (CreateAsSelect) statement);
+    }
+    final List<QueryMetadata> queryMetadataList =
+        ksqlEngine.buildMultipleQueries(statementString, ksqlConfig, configProperties);
+    if (queryMetadataList.size() != 1
+        || !(queryMetadataList.get(0) instanceof PersistentQueryMetadata)) {
+      throw new KsqlException("Could not build the query: " + statementString);
+    }
+    queryMetadataList.get(0).start();
+  }
+
+  private Query getQueryFromStatement(final String statementString, final Statement statement) {
+    if (statement instanceof CreateAsSelect) {
+      return  ((CreateAsSelect) statement).getQuery();
+    } else if (statement instanceof InsertInto) {
+      return  ((InsertInto) statement).getQuery();
+    } else {
+      throw new KsqlException("Only CSAS/CTAS and INSERT INTO are persistent queries: "
+          + statementString);
+    }
+  }
+
+  private void validateCsasCtas(final QueryMetadata queryMetadata,
+      final CreateAsSelect createAsSelect) {
+    if (createAsSelect instanceof CreateStreamAsSelect
+        && queryMetadata.getDataSourceType() == DataSource.DataSourceType.KTABLE) {
+      throw new KsqlException("Invalid result type. Your SELECT query produces a TABLE. "
+          + "Please use CREATE TABLE AS SELECT statement instead.");
+    } else if (createAsSelect instanceof CreateTableAsSelect
+        && queryMetadata.getDataSourceType() == DataSourceType.KSTREAM) {
+      throw new KsqlException("Invalid result type. Your SELECT query produces a STREAM. "
+          + "Please use CREATE STREAM AS SELECT statement instead.");
     }
   }
 
   private static String readQueriesFile(final String queryFilePath) {
-    final StringBuilder sb = new StringBuilder();
-    try (BufferedReader br = new BufferedReader(new InputStreamReader(
-        new FileInputStream(queryFilePath), StandardCharsets.UTF_8))) {
-      String line = br.readLine();
-      while (line != null) {
-        sb.append(line);
-        sb.append(System.lineSeparator());
-        line = br.readLine();
-      }
+    try {
+      return new String(java.nio.file.Files.readAllBytes(
+          Paths.get(queryFilePath)), StandardCharsets.UTF_8);
+
     } catch (IOException e) {
-      throw new KsqlException("Could not read the query file. Details: " + e.getMessage(), e);
+      throw new KsqlException(
+          String.format("Could not read the query file: %s. Details: %s",
+              queryFilePath, e.getMessage()),
+          e);
     }
-    return sb.toString();
+  }
+
+  public Map<String, Object> getConfigProperties() {
```

Comment:
If you return `Map<String, ?>` then the map is essentially immutable, (aside from someone casting it, obvs)

Reply:
returning `Map<String, ?>` now.